### PR TITLE
Augment match_clawback_puzzle to accept both types of programs

### PR DESCRIPTION
**Pull request by AmineKhaldi**
`match_clawback_puzzle` takes puzzles and solutions, and fetches conditions by passing those to `conditions_for_solution` which accepts both types of programs.

This is part of an effort to use serialized programs without the need to convert them into programs.

### CODE REVIEW

File: chia/wallet/puzzles/clawback/drivers.py [hunk 1]
```diff
@@ -121,17 +122,15 @@ def create_merkle_solution(
...
+    inner_solution: Union[Program, SerializedProgram],
 ) -> Optional[ClawbackMetadata]:
     # Check if the inner puzzle is a P2 puzzle
     if MOD != uncurried.mod:
         return None
     # Fetch Remark condition
-    conditions = conditions_for_solution(
-        inner_puzzle,
-        inner_solution,
-        DEFAULT_CONSTANTS.MAX_BLOCK_COST_CLVM // 8,
```
Comment (arvidn) [inline chia/wallet/puzzles/clawback/drivers.py:133]: generally I think you should avoid making unrelated changes to PRs

Comment (AmineKhaldi) [reply chia/wallet/puzzles/clawback/drivers.py:133]: I see this as related to my change. I altered the type of `inner_puzzle` and `inner_solution` in this line, and I'd like to read it as one line.

File: chia/wallet/puzzles/clawback/drivers.py [hunk 2]
```diff
@@ -121,17 +122,15 @@ def create_merkle_solution(
 
 
 def match_clawback_puzzle(
```
Comment (arvidn) [inline chia/wallet/puzzles/clawback/drivers.py:124]: I would expect some callers of this function to not have full mypy checking. Is that a concern of yours?
I haven't looked into it deeply, but `wallet_tools` and `did_wallet` looks like risks

Comment (AmineKhaldi) [reply chia/wallet/puzzles/clawback/drivers.py:124]: This is part of the type checking effort. If you have specific cases we need to check, please let us know.

**Review by arvidn — APPROVED** [review]
